File: backend/src/main/java/org/example/backend/analysis/service/AnalysisService.java
package org.example.backend.analysis.service;

import org.example.backend.analysis.dto.request.AnalysisRequest;
import org.example.backend.analysis.dto.response.AnalysisDetailResponse;
import org.example.backend.analysis.dto.response.AnalysisResponse;

public interface AnalysisService {

    AnalysisResponse analyzeProject(AnalysisRequest request);

    AnalysisDetailResponse getAnalysis(Long id);
}

File: frontend/src/app/features/analysis/services/analysis.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { AnalysisRequest } from '../../../features/analysis/models/analysis-request.model';
import { AnalysisResponse } from '../../../features/analysis/models/analysis-response.model';

@Injectable({
  providedIn: 'root'
})
export class AnalysisService {

  private http = inject(HttpClient);

  private readonly apiUrl = '/api/analyse';

  analyze(request: AnalysisRequest): Observable<AnalysisResponse> {

    return this.http.post<AnalysisResponse>(
      this.apiUrl,
      request
    );

  }

}
